ci-operator/step-registry/openshift-qe/rosa-loaded-upgrade/openshift-qe-rosa-loaded-upgrade-commands.sh

```diff
@@ -0,0 +1,180 @@
+
+  if [[ -z "$RELEASE_IMAGE_LATEST" ]]; then
+    echo "RELEASE_IMAGE_LATEST is an empty string, exiting"
+    exit 1
+  fi
+  echo RELEASE_IMAGE_LATEST is $RELEASE_IMAGE_LATEST
+
+  if [[ -z $RELEASE_IMAGE_INTERMEDIATE && -s "${SHARED_DIR}/perfscale-override-upgrade" ]]; then
+      RELEASE_IMAGE_INTERMEDIATE="$(< "${SHARED_DIR}/perfscale-override-upgrade")" 
+  fi
+  echo RELEASE_IMAGE_INTERMEDIATE is $RELEASE_IMAGE_INTERMEDIATE
+
+  IF_MINOR_Z_UPGRADE=${IF_MINOR_Z_UPGRADE:=true}
+  if [[ ${IF_MINOR_Z_UPGRADE} == "true" ]];then
+      TARGET_RELEASES="$(oc adm release info "${RELEASE_IMAGE_INTERMEDIATE}" --output=json | jq -r '.metadata.version')"
+  elif [[ ${IF_MINOR_Z_UPGRADE} == "false" ]];then
+      TARGET_RELEASES="$(oc adm release info "${RELEASE_IMAGE_LATEST}" --output=json | jq -r '.metadata.version')"
+  else
+      echo "Invalid value of IF_MINOR_Z_UPGRADE, only support true or false"
+      exit 1
+  fi
+
+  if [[ -z $CLUSTER_ID && -s "${SHARED_DIR}/cluster-id" ]];then
+       CLUSTER_ID=$(cat ${SHARED_DIR}/cluster-id)
+  fi
+
+  echo "######################################################################"
+  rosa list upgrade -c $CLUSTER_ID
+  echo "######################################################################"
+
+  RECOMMEND_VERSION=`rosa list upgrade -c $CLUSTER_ID | grep recommended | awk '{print $1}'`
+
+  #If fail to find TARGET_RELEASES, use RECOMMEND_VERSION
+  if [[ -z $TARGET_RELEASES ]];then
+	  UPGRADE_TO_VERSION=$RECOMMEND_VERSION
+  else
+	  UPGRADE_TO_VERSION=$TARGET_RELEASES
+  fi
+  echo  "-------------------------------------------------------------------------------------------"
+  echo  Loaded Upgrade from [ $CURRENT_VERSION ] to [ $UPGRADE_TO_VERSION ] for $CLUSTER_ID on $REGION
+  echo  "-------------------------------------------------------------------------------------------"
+  echo rosa upgrade cluster -c $CLUSTER_ID --mode=auto --region $REGION --version $UPGRADE_TO_VERSION  --schedule-date $SCHEDULE_DATE --schedule-time $SCHEDULE_TIME -y
+  echo "###############################`date`#######################################"
+  rosa upgrade cluster -c $CLUSTER_ID --mode=auto --region $REGION --version $UPGRADE_TO_VERSION --schedule-date $SCHEDULE_DATE --schedule-time $SCHEDULE_TIME -y
+
+
+  echo "Checking ROSA upgrade status ..."
+  INIT=1
+  MAXRETRY=120
+  while true
+  do
+	  UPGRADE_STATE=$(rosa describe upgrade --region $REGION -c $CLUSTER_ID | grep 'Upgrade State:' | awk -F':' '{print $2}' | tr -d ' ')
+	  if [[ $UPGRADE_STATE == "scheduled" ]];then
+                  echo "######################################################################"
+		  echo "ROSA Upgrade has been sucessfully scheduled"
+                  echo "######################################################################"
+		  break
+	  fi
+	  INIT=$(( $INIT + 1 ))
+
+	  if [[ $INIT -gt $MAXRETRY ]];then
+		  echo "Fail to schedule upgrade in limited time"
+		  exit 1
+	  fi
+	  sleep 10
+  done
+
+  INIT_NUM=1
```

Comment:
@liqcui My apologies. Not sure I understand the logic from here until end of below while loop. Is this logic to wait after Upgrade is already scheduled? If yes, can we add more details to the echo statement such as progressing from `scheduled` to next stage etc.. or whatever the stage name is?

Reply:
Actually only scheduled and started state, I remove this part and add more time to next checking STARTED loop